### DISCUSSION TITLE
How to persist scroll position on data change?

### DISCUSSION THREAD

**Discussion by Wordllban** [TanStack/virtual#859] [Q&A]
I'm receiving update from websocket which affects data that is rendered in virtual list
My elements are dynamic(using measureElement) and might change their height when click on them, like accordion
Each time I receive update from websocket and my data changes, the scroll position is reset. Most of the cases this is the same items, but with couple different values.
Is there possible way to avoid this behaviour and keep scroll position on such changes?

**Comment by piecyk** [comment]
Yes, it’s possible! You’ll need to keep the existing data until the new data arrives. However, it's hard to give a precise solution without seeing a small example. If you can create one, I’d be happy to help further.
